lwm2m-client/src/main.c

```diff
@@ -0,0 +1,452 @@
+#include "sol-log.h"
+
+#include "sol-lwm2m.h"
+#include "soletta.h"
+#include "sol-mainloop.h"
+#include "sol-vector.h"
+#include "sol-util.h"
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <errno.h>
+#include <time.h>
+
+#define LOCATION_OBJ_ID (6)
+#define LOCATION_OBJ_LATITUDE_RES_ID (0)
+#define LOCATION_OBJ_LONGITUDE_RES_ID (1)
+#define LOCATION_OBJ_TIMESTAMP_RES_ID (5)
+
+#define ONE_SECOND (1000)
+#define LIFETIME (60)
+
+#define SERVER_OBJ_ID (1)
+#define SERVER_OBJ_SHORT_RES_ID (0)
+#define SERVER_OBJ_LIFETIME_RES_ID (1)
+#define SERVER_OBJ_BINDING_RES_ID (7)
+#define SERVER_OBJ_REGISTRATION_UPDATE_RES_ID (8)
+
+#define SECURITY_SERVER_OBJ_ID (0)
+#define SECURITY_SERVER_SERVER_URI_RES_ID (0)
+#define SECURITY_SERVER_IS_BOOTSTRAP_RES_ID (1)
+#define SECURITY_SERVER_SERVER_ID_RES_ID (10)
+
+extern char _sheap;                 /* start of the heap */
+extern char *heap_top;
+
+struct location_obj_instance_ctx {
+    struct sol_timeout *timeout;
+    struct sol_lwm2m_client *client;
+    char *latitude;
+    char *longitude;
```

Comment:
You can change `sol_lwm2m_resource_init()`, so it creates a copy of the string. Then you don't need to store as string.